### DISCUSSION TITLE
%mem1% and other "%xxx%" don't replaced by it values

### DISCUSSION THREAD

**Discussion by medenlin** [arendst/Tasmota#19711] [Q&A] [answered]
Hi!
I have a trouble with %% expressions. Tasmota don't replace %xxx% expressions by it values:

17:40:39.898 CMD: mem1
17:40:39.903 RSL: RESULT = {"Mem1":""}
17:40:50.068 CMD: mem1 1800
17:40:50.073 RSL: RESULT = {"Mem1":"1800"}
17:40:59.555 CMD: var1 %mem1%
17:40:59.561 RSL: RESULT = {"Var1":"%mem1%"}
17:41:05.721 CMD: var1
17:41:05.726 RSL: RESULT = {"Var1":"%mem1%"}

I try to full reset config with no result.
What could this be due to?

**Comment by sfromis** [comment, marked as answer]
You can only use the **rule** variables in rules, not when commands are issued by other means.
